src/boimmgpy/model_annotation/model_annotator.py

```diff
@@ -0,0 +1,382 @@
+        if lipid_id != None and not lipid_id[2]:
+            for backbone in lipid_id[0]:
+                lipid_compound = self.get_coumpound(session,backbone, lipid_id[1], is_compound)
+                if lipid_compound != None and lipid_compound[0] != 0:
+                    structurally_defined_lipids = (
+                        self.get_compounds_with_specific_parent_set_of_components(
+                        session,lipid_compound[0], lipid_compound[1]
+                        )
+                    )
+                    if structurally_defined_lipids:
+                        if metabolite.id in results:
+                            results[metabolite.id] = (
+                                results[metabolite.id]
+                                + structurally_defined_lipids
+                            )
+                        else:
+                            results[metabolite.id] = structurally_defined_lipids
+                        counter[self.backbone] += 1
+
+                if lipid_compound != None and lipid_compound[0] == 0:
+                    backbone_compound = self.get_compound_from_synonym(session,backbone)
+                    structurally_defined_lipids = (
+                        self.get_compounds_with_specific_parent_set_of_components(session,
+                            backbone_compound, lipid_compound[1]
+                        )
+                    )
+                    if structurally_defined_lipids:
+                        if metabolite.id in results:
+                            results[metabolite.id] = (
+                                results[metabolite.id]
+                                + structurally_defined_lipids
+                            )
+                        else:
+                            results[metabolite.id] = structurally_defined_lipids
+                        counter[self.backbone] += 1
+
+        if lipid_id != None and lipid_id[2] == True:
+            is_compound = True
+            for backbone in lipid_id[0]:
+                lipid_compound = self.get_coumpound(session,backbone, lipid_id[1], is_compound)
+                if lipid_compound != None:
+                    structurally_defined_lipids = (
+                        self.get_compounds_with_specific_parent_set_of_components(
+                            session,lipid_compound[0], lipid_compound[1]
+                        )
+                    )
+                    if structurally_defined_lipids:
+                        if metabolite.id in results:
+                            results[metabolite.id] = (
+                                results[metabolite.id]
+                                + structurally_defined_lipids
+                            )
+                        else:
+                            results[metabolite.id] = structurally_defined_lipids
+                        counter[self.backbone] += 1
+        
+        
+        return counter,results
```

Comment:
I found it very difficult to read this method because of the ifs inside of if and for loops. moreover, it is quite difficult at the beginning of the code to understand what **lipid_id[0], lipid_id[1] ...** are. it would be way easier for the ones (and Adriano of the future) reading your code. In this sense, I think you must set the name of the variables **lipid_id[0], lipid_id[1] ...** at the beginning of the method and right after the **lipid_id = self.get_synonym_id(session,self.backbone, side_chain)**, or even passing directly as the output of the function like: **backbone_id, sidechain_id, compound_exists = self.get_synonym_id(session,self.backbone, side_chain)** and then use them in the rest of the function. Also for **lipid_compound = self.get_coumpound(session,backbone, lipid_id[1], is_compound)**

Finally, I will ask you to split the method into two: the first to do the first if and another to do the second.